### DISCUSSION TITLE
add custom css, js with webpack encore in different pages

### DISCUSSION THREAD

**Discussion by butachan** [symfony/symfony#47250] [Q&A]
Hello,
I d like to know the best practice (in terms of quality and performance) to add css and js when i am using webpack encore.

First, I don't know where should I put my files: in _assets/styles_ or in _public/builds_ ?
Then in my base.html.twig, I already have `{{ encore_entry_link_tags('app') }}` and it seems to refer to both _assets/styles/app.css_ and  _public/builds/app.css_. That's why I am confused.

Then, I want to add other css (and js) to specific pages. I tried first add my test.css in public/builds and assets/styles and I add
 {{ encore_entry_link_tags('test') }} in the twig which returns error 
>  ("Could not find the entry "test" in "../public/build/entrypoints.json". Found: app, admin."). 

but when I remove  {{ encore_entry_link_tags('test') }} from the twig, it works but in every page (as if my test.css content was in app.css)

How do I add css and js files in current page ?

**Comment by MatTheCat** [comment]
Hi!

> First, I don't know where should I put my files: in assets/styles or in public/builds ?

*Your* files are not supposed to already be “built” and thus they don’t need to be public, so you wouldn’t put them in `public/build`. Webpack Encore uses the `assets` folder as default root so you’d better use it too.

> Then, I want to add other css (and js) to specific pages.

Read https://symfony.com/doc/current/frontend/encore/simple-example.html#multiple-entries
